### PR TITLE
Add more metadata to the user agent

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds a bit more metadata to the user agent to allow us to have more statistics on the usage, more precisely, it registers:
- the type of the file asked on the hub ("config", "tokenizer", "model" or "model_card")
- the framework used for the model ("pytorch", "tensorflow" or "flax"). Note that this is the framework actually used even in the case of a conversion (so if download the PyTorch checkpoint but use it to instantiate a Flax model, it will be "flax")
- for a tokenizer, whether it's fast or slow (like from the framework it checks the class used at the end, not the files downloaded)
- whether the Auto API was used or not
- if the instantiation came from a given pipeline or not
- if the instantiation came from the CI or not (by using a specific env variable)

There is no personal data collected but if a user wants to deactivate this behavior, the `DISABLE_TELEMETRY` env variable can be set to any truthy value and none of this will be shared.

### CODE REVIEW

File: src/transformers/file_utils.py [hunk 1]
```diff
@@ -1227,13 +1227,18 @@ def http_user_agent(user_agent: Union[Dict, str, None] = None) -> str:
     """
     Formats a user-agent string with basic info about a request.
     """
+    if DISABLE_TELEMETRY:
+        return "telemetry/off"
```
Comment (n1t0) [inline src/transformers/file_utils.py]: Can we at least keep `transformers/version; telemetry/off`? Our analytics system only parses this if the transformers version comes first.

File: .github/workflows/self-push.yml [hunk 2]
```diff
@@ -44,6 +44,7 @@ jobs:
           OMP_NUM_THREADS: 8
           MKL_NUM_THREADS: 8
           HF_HOME: /mnt/cache
+          TRANSFORMERS_IS_CI: yes
```
Comment (n1t0) [inline .github/workflows/self-push.yml:47]: You can set this at the top level to specify it only once (cf https://github.com/huggingface/tokenizers/blob/master/.github/workflows/python-release.yml#L8)

**Review by n1t0 — APPROVED** [review]
LGTM, thanks you for adding this!

**Review by LysandreJik — APPROVED** [review]
LGTM

File: .github/workflows/self-push.yml (hunk 2 above)
Comment (julien-c) [reply .github/workflows/self-push.yml]: maybe we can also do that for Circle CI for consistency?

**Review by julien-c — APPROVED** [review]
Looks great!

**Pull request merged**